.eslintrc.js

```diff
@@ -62,48 +63,48 @@ module.exports = {
     //************************************************************************
     // Enforced
     //
+    curly: "error",
     // Set console calls to emit warnings as not enabled by eslint:recommended
     "no-console": "error",
     // Increase react-hooks/exhaustive-deps from warning to error
     "react-hooks/exhaustive-deps": "error",
-    // //************************************************************************
-    // // Relaxed: rules we are progressively adopting
-    // //
-    // // enforce types
-    // "@typescript-eslint/no-explicit-any": "warn",
-    // // enforce typed exports
-    // "@typescript-eslint/explicit-module-boundary-types": "warn",
-    // // enforce clean code
-    // "@typescript-eslint/no-unused-vars": "warn",
-    // // enforce interface naming convention
-    // "@typescript-eslint/naming-convention": [
-    //   "warn",
-    //   {
-    //     "selector": "interface",
-    //     "format": ["PascalCase"],
-    //     "custom": {
-    //       "regex": "^I[A-Z]",
-    //       "match": true
-    //     }
-    //   },
-    //   {
-    //     "selector": "typeAlias",
-    //     "format": ["PascalCase"],
-    //     "custom": {
-    //       "regex": "^T[A-Z]",
-    //       "match": true
-    //     }
-    //   }
-    // ],
-    // // enforce import ordering
-    // "sort-imports": [
-    //   "warn",
-    //   {
-    //     "allowSeparatedGroups": true,
-    //     "memberSyntaxSortOrder": ["none", "all", "single", "multiple"],
-    //     "ignoreCase": false
-    //   }
-    // ],
+    // enforce types
+    "@typescript-eslint/no-explicit-any": "error",
+    // Bans specific types from being used (e.g. builtin types)
+    "@typescript-eslint/ban-types": "error",
+    // enforce typed exports
+    "@typescript-eslint/explicit-module-boundary-types": "error",
+    // enforce clean code
+    "@typescript-eslint/no-unused-vars": "error",
+    // enforce interface naming convention
+    "@typescript-eslint/naming-convention": [
+      "error",
+      {
+        selector: "interface",
+        format: ["PascalCase"],
+        custom: {
+          regex: "^I[A-Z]",
```

Comment:
Would it be sufficient to use [A-Z] for interfaces and types? The reason is that interfaces and types are the same in Typescript and can be used interchangeably (except few edge and rare cases). That's why I'm wondering if it makes sense to treat them equally or we can instead encourage the use of Type Alias instead, so we don't have mixed use of both (I find Type conciser and more flexible especially when creating unions/intersections).

Reply:
Surely if you do that it will also prevent having issues with cypress since all the cypress interfaces do not start with I

Reply:
Having a prefix is quite handy when you import Types or Interfaces, e.g. to differentiate from Classes if we were to use only [A-Z]?

Reply:
I haven't felt the need to use distinct formats for classes and interfaces in the past as it's usually clear from the context., but it might be a matter of preference :) The only concern was that the differentiation between interfaces and types might create the impression they are to be treated differently despite the fact they can be used interchangeably in most cases.

Reply:
While I would prefer being able to differentiate types/interface from classes (assuming we don't have functions or objects starting with an uppercase), I think we should define the guidelines for this kind of usage first. As we said, currently, we use types and interface interchangeably, which does not help in this case.

Rather than enforcing the naming, would it make sense to recommend one or the other?

I would not dwell on this one: it is more important to already apply all the other rules as soon as possible.

Reply:
ok, reverted this change for now until there is consensus ;)